### ISSUE TITLE
from_dict fails when Dict is bare.

### ISSUE THREAD

**Issue by jcmuddle** [lidatong/dataclasses-json#232]
dataclasses-json __version__ = 0.5.1
Python __version__ = 3.6.10, 3.7.7,  and 3.8.3

When defining a dataclass with Dict typing which is bare, the from_dict method fails.

```
from typing import Dict

from dataclasses import dataclass
from dataclasses_json import DataClassJsonMixin


@dataclass
class Minimal(DataClassJsonMixin):

    test: Dict


if __name__ == "__main__":
    test_dict = Minimal(test={"foo": "bar"}).to_dict()
    test = Minimal.from_dict(test_dict)
```

Running the above gives the following stack trace on Python 3.6
```
Traceback (most recent call last):
  File "minimal.py", line 15, in <module>
    test = Minimal.from_dict(test_dict)
  File "/home/jmuddle/.conda/envs/minimal_3.6/lib/python3.6/site-packages/dataclasses_json/api.py", line 83, in from_dict
    return _decode_dataclass(cls, kvs, infer_missing)
  File "/home/jmuddle/.conda/envs/minimal_3.6/lib/python3.6/site-packages/dataclasses_json/core.py", line 200, in _decode_dataclass
    elif _is_supported_generic(field_type) and field_type != str:
  File "/home/jmuddle/.conda/envs/minimal_3.6/lib/python3.6/site-packages/dataclasses_json/core.py", line 238, in _is_supported_generic
    type_) or is_union_type(type_) or is_enum
  File "/home/jmuddle/.conda/envs/minimal_3.6/lib/python3.6/site-packages/dataclasses_json/utils.py", line 92, in _is_optional
    _hasargs(type_, type(None)) or
  File "/home/jmuddle/.conda/envs/minimal_3.6/lib/python3.6/site-packages/dataclasses_json/utils.py", line 49, in _hasargs
    res = all(arg in type_.__args__ for arg in args)
  File "/home/jmuddle/.conda/envs/minimal_3.6/lib/python3.6/site-packages/dataclasses_json/utils.py", line 49, in <genexpr>
    res = all(arg in type_.__args__ for arg in args)
TypeError: argument of type 'NoneType' is not iterable
```

On Python 3.7 and 3.8 the above gives the following stack trace
```
  File "/home/jmuddle/.local/lib/python3.7/site-packages/dataclasses_json/core.py", line 263, in _decode_generic
    res = _get_type_cons(type_)(xs)
TypeError: 'TypeVar' object is not callable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "minimal.py", line 15, in <module>
    test = Minimal.from_dict(test_dict)
  File "/home/jmuddle/.local/lib/python3.7/site-packages/dataclasses_json/api.py", line 83, in from_dict
    return _decode_dataclass(cls, kvs, infer_missing)
  File "/home/jmuddle/.local/lib/python3.7/site-packages/dataclasses_json/core.py", line 203, in _decode_dataclass
    infer_missing)
  File "/home/jmuddle/.local/lib/python3.7/site-packages/dataclasses_json/core.py", line 265, in _decode_generic
    res = type_(xs)
  File "/var/miniconda3/lib/python3.7/typing.py", line 673, in __call__
    raise TypeError(f"Type {self._name} cannot be instantiated; "
TypeError: Type Dict cannot be instantiated; use dict() instead
```

The solution is to do the following:
```
@dataclass
class Minimal(DataClassJsonMixin):

    test: Dict[str, str]
```
My expectation would be that an error is raised which is a little more informative or that a bare Dict type works.

**Comment by advance512** [comment]
Can confirm, this was a problem for us as well.
Thanks for the workaround, @jcmuddle

**Comment by matt035343** [comment]
I have been trying to dig into this a bit, and I was able to reproduce in Python 3.8 / dataclasses-json 0.5.7.

When debugging, it is very funky behaviour which I suspect is a bug in Python itself, because it have no issues in Python 3.9.

Basically, the issue on following line:
https://github.com/lidatong/dataclasses-json/blob/09f43c6eee00411264500aef475381f4ee3f4db2/dataclasses_json/core.py#L275
The function returns the correct result, which is `type_.__origin__`.  However, calling `type_.__origin__(xs)` (or even without the argument `xs`), the error `TypeError: 'TypeVar' object is not callable` is raised. The second time you call the exact same function, no error is raised and the program continues.

**Comment by george-zubrienko** [comment]
i'd say just add a check of a python version from `sys` and for stuff earlier than 3.9 do the hack, comment accordingly :)

**Comment by matt035343** [comment]
This issue seems to have fixed itself in v0.5.8. Probably by https://github.com/lidatong/dataclasses-json/pull/346 or https://github.com/lidatong/dataclasses-json/pull/394.

Please reopen if the issue persists
